### PR TITLE
Drive space fix

### PR DESCRIPTION
**Pull request by mehdibalouchi**
Add DriveSpace.Save
Add correct usage of publish and save

### CODE REVIEW

File: drive/drive.go [hunk 1]
```diff
@@ -51,7 +51,8 @@ func (ud *UserDrive) PublicSpace(ctx context.Context, api fxiface.CoreAPI) (*Dri
 		Api:       api,
 		SpaceType: PublicDriveSpaceType,
 		RootCid:   ud.PublicSpaceCid,
-		RootDir:   rootDir.(files.Directory)}}, err
+		RootDir:   rootDir.(files.Directory),
+		Drive:     ud}}, err
```
Comment (masih) [inline drive/drive.go:55]: does this code compile?

File: drive/drive-space.go [hunk 2]
```diff
@@ -46,6 +46,7 @@ type DriveSpace struct {
 	SpaceType DriveSpaceType
 	RootCid   string
```
Comment (ehsan6sha) [inline drive/drive-space.go:47]: RootCid is the root folder? For example "Ehsan" in a folder structure of: Ehsan->Photos->Summer2020

Comment (mehdibalouchi) [reply drive/drive-space.go:47]: @ehsan6sha  Each `UserDrive` has two `DriveSpace`s (PublicSpace and PrivateSpace). Each space keeps it's own rootDir CID and a pointer to it's parent drive. All of changes happen on the spaces. After you are done changing a driveSpace (with methods like write, mkdir, ...) you should call DriveSpace.Save() which will update the parent drive (according the spaceType).
After all of it, when you are done with the drive, you call UserDrive.Publish() which publishes the drive on the DriveStore

File: drive/drive.go (hunk 1 above)
Comment (mehdibalouchi) [reply drive/drive.go:55]: yes, why?

Comment (masih) [reply drive/drive.go:55]: I see the formatting made that line seem odd to me.

**Review by masih — APPROVED** [review]

**Pull request merged**